internal/domain/permission.go

```diff
@@ -39,3 +39,5 @@ const (
 	PermissionIDPRead             = "iam.idp.read"
 	PermissionOrgIDPRead          = "org.idp.read"
 )
+
+type ApplicationPermissionCheck func(ctx context.Context, clientID, userID string) (err error)
```

Comment:
Unsure if still needed on domain level or just command package if moved there.
And as remarked earlier, "ApplicationPermission" to me refers to permissions for the application, but the check ensures some preconditions on the authentication

Reply:
I named it application permissions, as there are other preconditions that could be check in the same step as well, and in the end the preconditions are application specific through the clientID and entityID.
I moved it to the domain package as the checkPermission is also defined here and provided from the API to the command and query package.

Reply:
Renamed it to ProjectPermissionCheck, but left in domain as it is used by SAML and OIDC on the command side

Reply:
For me the `Permission` part still is a bit off... there's no actual permission involved, but rather an additional condition
Also unsure then if we really should return a PermissionDeniedError based on this check. To me this indicate the caller does not have the permission, but it's acutally related to the authenticated user. On the other hand precondition error feels also not correct.

Reply:
Added comment regarding usage, relations and configuration of this check.